python/spinn/util/chainer_blocks.py

```diff
@@ -226,6 +275,31 @@ def slice_gate(gate_data, i):
     return (c_t, h_t)
 
 
+def treegru(h_left, h_right, gates, U_left, U_right):
```

Comment:
`h_left` and `h_right` are either word embeddings if performing on the leaves, or the previous h's. Gates is something like `Linear(left) + Linear(right)`. `U_left` and `U_right` are instances of `Linear`.